src/Transformation/Transformation.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/* Copyright (c) 2017 Richard Klees <richard.klees@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Transformation;
+
+/**
+ * A transformation is a function from one datatype to another.
+ *
+ * It MUST NOT perform any sideeffects, i.e. it must be morally impossible to observe
+ * how often the transformation was actually performed. It MUST NOT touch the providedi
+ * value, i.e. it is allowed to create new values but not to modify existing values.i
+ * This would be an observable sideeffect.  
```

Comment:
This guideline-like information is redundant. I would prefer a short reference to the README for more information here. Otherwise we risk, that one of the guidelines will be outdated soon.